src/components/cl/hier/alltoallv/alltoallv.c

```diff
@@ -250,10 +272,16 @@ UCC_CL_HIER_PROFILE_FUNC(ucc_status_t, ucc_cl_hier_alltoallv_init,
     *task = &schedule->super;
     return UCC_OK;
 
+err_init_3:
+    if (!full_only) {
+        ucc_collective_finalize(&task_node->super);
+    }
 err_init_2:
     ucc_collective_finalize(&task_full->super);
 err_init_1:
-    ucc_mc_free(cl_schedule->scratch);
+    if (!full_only) {
```

Comment:
minor: if line 182 send to `error` instead of `err_init_1`, we can remove this branching.
Also we can completely get rid of `full_only`

Reply:
needed for the same case as explained in previous free comment